src/imagery/i.landsat/i.landsat.import/i.landsat.import.py

```diff
@@ -159,7 +159,25 @@ def _untar(inputdir, untardir):
 
     for scene in scenes_to_untar:
         with tarfile.open(name=scene, mode="r") as tar:
-            tar.extractall(untardir)
+
+            def is_within_directory(directory, target):
+
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+
+                return prefix == abs_directory
+
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+                for member in tar.getmembers():
+                    member_path = os.path.join(path, member.name)
+                    if not is_within_directory(path, member_path):
+                        raise Exception(_("Attempted path traversal in tar file"))
```

Comment:
```suggestion
                        raise RuntimeError(_("Attempted path traversal in tar file"))
```
I'm settling for `RuntimeError` as for this one I'm convinced it is an appropriate Python exception for this case and I understand the reach of it. For the other option, `fatal`, I don't understand enough the reach to know if it is more appropriate or not. If needed, it can be changed at a later time in another PR.

Reply:
This will end the needed reviews and will be finally ready to merge